### PR TITLE
ENH: stats.kde: add `data_covariance` keyword argument

### LINKED ISSUE #19250: ENH: support a fixed bandwidth in multivariate gaussian_kde
### Is your feature request related to a problem? Please describe.

At the moment, multivariate `gaussian_kde` selects a bandwidth automatically for each dimension separately. While this makes sense for a lot of use cases, it cannot be applied to coordinates representing points on a plane. This is different than what sklearn is doing in neighbors.KernelDensity, where a single bandwidth is used across all dimensions.

### Describe the solution you'd like.

I would like to have an option to specify not only the factor of the bandwidth but fix the bandwidth across all dimensions to a set scalar (i.e. meters in case of geographic points with x, y coordinates used as variables). As far as I can tell, there is no way achieving that right now.

### Describe alternatives you've considered.

I can use sklearn directly, which is often fine. However, the issue arised in seaborn (https://github.com/mwaskom/seaborn/issues/3472) which wraps `gaussian_kde` in its `kdeplot` and where adding sklearn as a dependency is a no-go.

### Additional context (e.g. screenshots, GIFs)

See the original issue in seaborn for visual representation of an issue and a comparison with a sklearn implementation -> https://github.com/mwaskom/seaborn/issues/3472

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-19250

#### What does this implement/fix?
This adds a `data_covariance` keyword argument to the initializer of `scipy.stats.kde`. This allows the user to pass the data covariance matrix if it is already known, avoiding redundant calculation, or to use a matrix other than the sample covariance (e.g. the identity, as in gh-19250).

#### Additional information
This needs an email to the mailing list before being merged.

### CODE REVIEW

**Review by tupui — APPROVED** [review]
LGTM, let's see if this is of interest for seaborn before asking the ML.

**Comment by mwaskom** [comment]
FWIW this strikes me as a very unintuitive parameterization. As I understand it (and I might not), you're exposing a covariance matrix parameter to essentially enable the fixed-bandwidth use-case in #19250 as a hack, where the user would pass an identity matrix to enforce a common bandwidth on all dimensions? (Which both makes the bandwidth the same on all dimensions and changes the interpretation of a scalar `bandwidth_method` from a scaling factor to a literal measurement of the bandwidth?)

Are there any first-order use-cases where a user would actually want to pass a (non-identity) covariance matrix that is different from what would be computed from their data? If not, maybe the proposed use-case could be supported more directly?

File: scipy/stats/_kde.py
Comment (tupui) [reply scipy/stats/_kde.py:600]: ark ok then 👍

**Comment by mdhaber** [comment]
This PR shows that gh-19250 is quite simple, technically; it just needs a champion to hammer out the interface. I'll close this but leave gh-19250 open.

**Comment by josef-pkt** [comment]
> Are there any first-order use-cases where a user would actually want to pass a (non-identity) covariance matrix that is different from what would be computed from their data? If not, maybe the proposed use-case could be supported more directly?

One of the main usecases would be to use an outlier robust covariance (e.g. from statsmodels after it gets merged)

AFAICS
bw_factor can be set to 1, so that user has full control over the chosen kernel matrix


disclaimer: I only looked at part of the code and might not remember correctly the rest.

**Pull request closed without merging**
